### PR TITLE
add support for tweens tagged with "event triggers"

### PR DESCRIPTION
**Pull request by alexm-sgn**

### CODE REVIEW

File: src/starlingbuilder/engine/tween/ITweenBuilder.as [hunk 1]
```diff
@@ -44,16 +44,18 @@ package starlingbuilder.engine.tween
...
         /**
          * Stop tween according tweenData in layout
          *
          * @param root of the DisplayObject needs to be tween
          * @param paramsDict A dictionary of the mapping of every UIElement to its layout data, if null then stop all the available tweens
          * @param names array of names (e.g. ["container.button1", "container.button2") to be stop tween, if null then stop all the available tweens
+         * @param trigger string used to run some of the tweens on an object, if null then start all available
          */
-        function stop(root:DisplayObject, paramsDict:Dictionary = null, names:Array = null):void;
+        function stop(root:DisplayObject, paramsDict:Dictionary = null, names:Array = null, trigger:String = null):void;
```
Comment (jdelvalle) [inline src/starlingbuilder/engine/tween/ITweenBuilder.as]: We should remove the trigger param, is not used.

File: src/starlingbuilder/engine/tween/DefaultTweenBuilder.as [hunk 2]
```diff
@@ -60,11 +60,21 @@ package starlingbuilder.engine.tween
                     if (tweenData is Array)
                     {
                         for each (var item:Object in tweenData)
-                            createTweenFrom(obj, item);
+                        {
+                            if (!trigger
```
Comment (jdelvalle) [inline src/starlingbuilder/engine/tween/DefaultTweenBuilder.as]: Lets write this in one line.
Also, the second condition can be replaced with item.triggerName == trigger.

File: src/starlingbuilder/engine/tween/DefaultTweenBuilder.as [hunk 3]
```diff
@@ -60,11 +60,21 @@ package starlingbuilder.engine.tween
...
+                                || (item.hasOwnProperty("triggerName") && item.triggerName == trigger))
+                            {
```
Comment (jdelvalle) [inline src/starlingbuilder/engine/tween/DefaultTweenBuilder.as]: remove curly braces

File: src/starlingbuilder/engine/tween/DefaultTweenBuilder.as [hunk 4]
```diff
@@ -60,11 +60,21 @@ package starlingbuilder.engine.tween
...
+                                createTweenFrom(obj, item);
+                            }
+                        }
                     }
                     else
                     {
-                        createTweenFrom(obj, tweenData);
+                        if (!trigger
```
Comment (jdelvalle) [inline src/starlingbuilder/engine/tween/DefaultTweenBuilder.as]: Lets write this in one line.
Also, the second condition can be replaced with tweenData.triggerName == trigger.

File: src/starlingbuilder/engine/tween/DefaultTweenBuilder.as [hunk 5]
```diff
@@ -43,9 +43,9 @@ package starlingbuilder.engine.tween
         /**
          * @inheritDoc
          */
-        public function start(root:DisplayObject, paramsDict:Dictionary, names:Array = null):void
+        public function start(root:DisplayObject, paramsDict:Dictionary, names:Array = null, trigger:String = null):void
         {
-            stop(root, paramsDict, names);
+            stop(root, paramsDict, names, trigger);
```
Comment (jdelvalle) [inline src/starlingbuilder/engine/tween/DefaultTweenBuilder.as]: Check the comment in ITweenBuilder.as

File: src/starlingbuilder/engine/tween/DefaultTweenBuilder.as [hunk 6]
```diff
@@ -115,7 +125,7 @@ package starlingbuilder.engine.tween
         /**
          * @inheritDoc
          */
-        public function stop(root:DisplayObject, paramsDict:Dictionary = null, names:Array = null):void
+        public function stop(root:DisplayObject, paramsDict:Dictionary = null, names:Array = null, trigger:String = null):void
```
Comment (jdelvalle) [inline src/starlingbuilder/engine/tween/DefaultTweenBuilder.as]: Check the comment in ITweenBuilder.as

File: src/starlingbuilder/engine/tween/DefaultTweenBuilder.as [hunk 7]
```diff
@@ -60,11 +60,21 @@ package starlingbuilder.engine.tween
...
+                            {
+                                createTweenFrom(obj, item);
+                            }
+                        }
                     }
                     else
                     {
-                        createTweenFrom(obj, tweenData);
+                        if (!trigger
+                            || (tweenData.hasOwnProperty("triggerName") && tweenData.triggerName == trigger))
```
Comment (jdelvalle) [inline src/starlingbuilder/engine/tween/DefaultTweenBuilder.as]: this condition can be replaced with tweenData.triggerName == trigger.

**Review by jdelvalle — CHANGES_REQUESTED** [review]

File: src/starlingbuilder/engine/tween/DefaultTweenBuilder.as [hunk 8]
```diff
@@ -60,11 +60,21 @@ package starlingbuilder.engine.tween
...
+                        {
```
Comment (jdelvalle) [inline src/starlingbuilder/engine/tween/DefaultTweenBuilder.as]: remove curly braces

**Review by jdelvalle — CHANGES_REQUESTED** [review]

File: src/starlingbuilder/engine/tween/ITweenBuilder.as (hunk 1 above)
Comment (alexm-sgn) [reply src/starlingbuilder/engine/tween/ITweenBuilder.as]: removed

**Pull request merged**